### ISSUE TITLE
Tusk forgets authorization for Google Drive

### ISSUE THREAD

**Issue by dennisfrett** [subdavis/Tusk#126]
## This issue is a 

- [x] bug
- [ ] feature
- [ ] question

### Please describe the current behavior, and explain why it's bad.
After a while, Tusk disables access to Google Drive and shows one of the following error messages: (Screenshots included)
```
Error: Error: Request failed with status code 401
```
```
Error: You must Authorize google drive access to continue.
```

If I re-enable Google Drive, I have to select the account I want to use again and re-authorize Tusk to access to it again through Google. Then everything works for at least a few minutes until it resets again.

### Please describe how you think it should change.
This seems like unintended behaviour and possible a bug with Tusk - Google Drive integration.

### Screenshots
<img width="823" alt="screen shot 2018-03-21 at 09 11 03" src="https://user-images.githubusercontent.com/659950/37704271-b4b1704c-2cf8-11e8-87d7-d275527bb58e.png">
<img width="834" alt="screen shot 2018-03-21 at 11 12 26" src="https://user-images.githubusercontent.com/659950/37704304-c27fd8c6-2cf8-11e8-9919-4882156a4150.png">

**Comment by subdavis** [comment]
1) What browser do you use?
2) If chrome, are you signed into chrome at the browser level?
3) Try intentionally removing permissions for Tusk @ https://myaccount.google.com/permissions?pli=1 and starting over.

**Comment by subdavis** [comment]
Also please note that https://github.com/subdavis/Tusk/issues/64 was a hard-fought issue.  I'd encourage you to read over that issue and try the troubleshooting steps discussed there.

Let me know if anything else interesting crops up.

**Comment by dennisfrett** [comment]
Hi, thanks for the fast reply! I probably missed issue #64 when searching.
I am using Google Chrome, I am logged into a few Google accounts in the browser, one of which is the Google Drive account.

I remember the issue started after I first changed the password of my Google Drive account.

It was solved by removing the Tusk App from the Google Drive, and re-add it though Tusk!

**Comment by dennisfrett** [comment]
I'm sorry. I've closed the issue to soon it seems.
It just took longer than usual for the issue to reoccur.

**Comment by subdavis** [comment]
Hmm, interesting that you have multiple accounts in the browser.  That makes my spider senses tingle.  

Is the account you use with Tusk your primary or some secondary?  I'd be very curious to know if the same problem persists with either:

1) Firefox, where google chrome can't screw with your accounts or
2) In an instance of chrome with only one active login.

I might sign up for another gmail account to test this out myself.

**Comment by dennisfrett** [comment]
It indeed seems to be the fact that it's a secondary account in Chrome. Whenever I restart the browser, the "Re-authorize" pop-up seems to have forgotten this account. When I just visit any google website, however, all my accounts are always listed. It is still unclear to me how I can force this account to stay logged in/connected.

**Comment by subdavis** [comment]
Bingo.

From the documentation:  https://developer.chrome.com/apps/identity#method-getAuthToken

for getAuthToken:

> Since Chrome 37.
> The account ID whose token should be returned. If not specified, the primary account for the profile will be used.  account is only supported when the "enable-new-profile-management" flag is set.

This is completely unhelpful for two reasons: 
* When I launch getAuthToken(), there is no way for Tusk to know what account the token came from so I can request from the same account next time.
* The getAccounts() API is not in mainline chrome, so I can't even get a list of the accounts you have loaded.

So to recap: I can 99% guarantee that this is the reason for your trouble.  This being the case, there's absolutely nothing I can do about it until Chrome provides reasonable APIs for profile management.  

What you can do:
* Make the Google Account you need for tusk your primary profile.
* Use firefox.

TL;DR Chrome bug, cannot fix.

**Comment by subdavis** [comment]
For anyone still struggling with this problem: https://github.com/subdavis/Tusk/wiki/Troubleshooting#google-drive-constant-reauthorization

Closing this issue and rolling it into #133
